### PR TITLE
JBPM-4682 Remote Workbench Performance Tests

### PR DESCRIPTION
**Pull request by ibek**

### CODE REVIEW

File: performance-tests/remotewb-performance-tests/addUsers.sh [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#!/bin/sh
+
+$JBOSS_HOME/bin/add-user.sh -a perfUser 'perfUser1234;'
+$JBOSS_HOME/bin/add-user.sh -a engUser 'engUser1234;'
+echo "perfUser=admin" >> $JBOSS_HOME/standalone/configuration/application-roles.properties
+echo "engUser=admin,engineering" >> $JBOSS_HOME/standalone/configuration/application-roles.properties
```
Comment (mrietveld) [inline performance-tests/remotewb-performance-tests/addUsers.sh]: You could probably do this with a maven plugin?

**Comment by mrietveld** [comment]
Hi Ivo, 

I'll merge this immediately: my comments are just some suggestions..

File: performance-tests/remotewb-performance-tests/src/main/resources/logback.xml [hunk 2]
```diff
@@ -0,0 +1,65 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (mrietveld) [inline performance-tests/remotewb-performance-tests/src/main/resources/logback.xml]: In general, we try to only use `logback-test.xml` files, and also exclude `logback.xml` files from the jars. 

Logback will use the first `logback.xml` or `logback-test.xml` file that it encounters in the classpath: this means that if this jar is in the classpath of another project, it will be impossible to override the settings in this file.

File: performance-tests/workbench-assets/deploy.sh [hunk 3]
```diff
@@ -0,0 +1,5 @@
+#/bin/sh
+mvn clean install deploy -s settings.xml
+echo "Setting up worbench-assets deployment ..."
+curl -X POST --header "Authorization: Basic cGVyZlVzZXI6cGVyZlVzZXIxMjM0Ow==" http://localhost:8080/business-central/rest/deployment/org.kie.perf:workbench-assets:1.0.0-SNAPSHOT/deploy?strategy=SINGLETON
+sleep 5
```
Comment (mrietveld) [inline performance-tests/workbench-assets/deploy.sh]: Maven plugin?

File: performance-tests/workbench-assets/src/main/java/.gitignore [hunk 4]
```diff
@@ -0,0 +1 @@
+# empty
```
Comment (mrietveld) [inline performance-tests/workbench-assets/src/main/java/.gitignore]: Is this file necessary?

File: performance-tests/workbench-assets/src/main/resources/.gitignore [hunk 5]
```diff
@@ -0,0 +1 @@
+# empty
```
Comment (mrietveld) [inline performance-tests/workbench-assets/src/main/resources/.gitignore]: Necessary?

File: performance-tests/workbench-assets/src/main/resources/processes/.gitignore [hunk 6]
```diff
@@ -0,0 +1 @@
+# empty
```
Comment (mrietveld) [inline performance-tests/workbench-assets/src/main/resources/processes/.gitignore]: Necessary?

**Comment by mrietveld** [comment]
Merged: 93d4bb3e6f3372ab2b582567eed4f64a501b92d4

File: performance-tests/remotewb-performance-tests/src/main/resources/logback.xml (hunk 2 above)
Comment (ibek) [inline performance-tests/remotewb-performance-tests/src/main/resources/logback.xml]: AFAIK this jar is not published to any maven repository. These modules contain only tests. Therefore, I do not think that it is realistic that somebody would use this module as a dependency. ;-)

File: performance-tests/remotewb-performance-tests/addUsers.sh (hunk 1 above)
Comment (ibek) [inline performance-tests/remotewb-performance-tests/addUsers.sh]: What maven plugin do you have in mind?

File: performance-tests/workbench-assets/src/main/java/.gitignore (hunk 4 above)
Comment (ibek) [inline performance-tests/workbench-assets/src/main/java/.gitignore]: you tell me :) this is generated by Business Central automatically - I have just copied the project from there. Since it is actually not necessary I will remove it in the next commit. :)

**Comment by ibek** [comment]
Thank you Marco for the check and merge.

File: performance-tests/remotewb-performance-tests/addUsers.sh (hunk 1 above)
Comment (mrietveld) [inline performance-tests/remotewb-performance-tests/addUsers.sh]: Something like [this](https://github.com/droolsjbpm/kie-tests/blob/master/remote-kie-tests/kie-wb-tests/kie-wb-tests-eap/pom.xml#L365-L384)? 

Basically, you generate the properties once, store them in a directory, and then copy them over the default ones using the resources plugin.

Comment (mrietveld) [inline performance-tests/remotewb-performance-tests/addUsers.sh]: BTW, the reason behind my thoughts is that it's always nice if people can simply run the whole test using "mvn install"... The easier it is to run, the more other developers will use and support the test.

Comment (ibek) [inline performance-tests/remotewb-performance-tests/addUsers.sh]: Ok, I will do something with it. However, deploy.sh will be more complicated when some processes may stuck and then it is not possible to redeploy the deployment unit.
